cmd/snap-update-ns/utils.go

```diff
@@ -100,20 +100,34 @@ func (sec *Secure) OpenPath(path string) (int, error) {
 	//  O_NOFOLLOW: don't follow symlinks
 	//  O_DIRECTORY: we expect to find directories
```

Comment:
It might make sense to re-write this as `O_DIRECTORY: we expect to find directories, except for the leaf item`